win: fix tests for nondebug config

wlemkows opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/pmem/nvml/1477)
<!-- Reviewable:end -->

lplewa commented:
Review status: 0 of 9 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/test/blk_pool_lock/blk_pool_lock.c, line 58 at r1](https://reviewable.io:443/reviews/pmem/nvml/1477#-KXfFSl2lDIQgqmH724Z:-KXfFSl39Kcq0o_H3UVG:b-th3vfh) ([raw file](https://github.com/pmem/nvml/blob/57ff0d4f273380293bd0b81f73223310f925df92/src/test/blk_pool_lock/blk_pool_lock.c#L58)):*
> ```C
> 
> 	/* start the 2nd test process */
> 	if (!CreateProcess(NULL,
> ```

 I think you miss parameter with path of the pool file

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1477)*
<!-- Sent from Reviewable.io -->

wlemkows commented:
Review status: 0 of 9 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/test/blk_pool_lock/blk_pool_lock.c, line 58 at r1](https://reviewable.io:443/reviews/pmem/nvml/1477#-KXfFSl2lDIQgqmH724Z:-KXj7IPWBIKc27aqZLSZ:b-896fix) ([raw file](https://github.com/pmem/nvml/blob/57ff0d4f273380293bd0b81f73223310f925df92/src/test/blk_pool_lock/blk_pool_lock.c#L58)):*
<details><summary><i>Previously, lplewa (Łukasz Plewa) wrote…</i></summary><blockquote>

 I think you miss parameter with path of the pool file
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1477)*
<!-- Sent from Reviewable.io -->

pbalcer commented on src/test/blk_pool_lock/blk_pool_lock.c:
```diff
@@ -184,9 +184,9 @@ main(int argc, char *argv[])
+		test_open_in_different_process(argv, 0);
 		for (int i = 1; i < 100000; i *= 2)
-			test_open_in_different_process(argv[1], i);
+			test_open_in_different_process(argv, i);
```
Instead of turning the variable named "path" to an array, add a second argument to those functions.

pbalcer commented on src/test/blk_pool_lock/blk_pool_lock.c:
```diff
@@ -45,18 +45,16 @@
-	MultiByteToWideChar(CP_ACP, 0, path, -1, parm, nChars);
-	_tcscat(cmd, parm);
-	_tcscat(cmd, L" X");
+	strcat(cmd, path[0]);
```
Might I suggest snprintf?

pbalcer commented on src/test/obj_pool_lock/obj_pool_lock.c:
```diff
@@ -45,18 +45,16 @@
  */
 int
-test_process(const char *path, int sleep)
+test_process(char **path, int sleep)
```
This is the third copy of this exact same code. My suggestion would be to move it to unittest.c, and while you are at it, change it so that it automatically retrieves the executable name so that you don't have to do anything else.

krzycz commented:
Reviewed 6 of 9 files at r1, 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[src/test/blk_pool_lock/blk_pool_lock.c, line 48 at r2](https://reviewable.io:443/reviews/pmem/nvml/1477#-KXzkvuJv4174AbM-sH3:-KXzkvuJv4174AbM-sH4:bkgayyt) ([raw file](https://github.com/pmem/nvml/blob/85b5694cb1427817897da9a33ab7936922b191cd/src/test/blk_pool_lock/blk_pool_lock.c#L48)):*
> ```C
>  */
> int
> test_process(char **path, int sleep)
> ```

1) I would just name it "argv".
2) Why not "const"?
3) To be honest, I'd convert into a generic function that would take "int argc, const char *argv[]" and would act as a utility function to spawn a new process with given args (fork + exec equivalent).

---

*[src/test/blk_pool_lock/blk_pool_lock.c, line 54 at r2](https://reviewable.io:443/reviews/pmem/nvml/1477#-KXfDhsFyVIvYXnxz-Uv-r2-54:-KXznMWx3FsK8RxewbFR:b-lsr6p8) ([raw file](https://github.com/pmem/nvml/blob/85b5694cb1427817897da9a33ab7936922b191cd/src/test/blk_pool_lock/blk_pool_lock.c#L54)):*
<details><summary><i>Previously, pbalcer (Piotr Balcer) wrote…</i></summary><blockquote>

Might I suggest snprintf?
</blockquote></details>

Because of comment 3) above, I think strcat() is more convenient, but it should be executed in a loop to concatenate all the args....
... unless we use a function that could take argv[] array.
I wonder if _spawnv() could do that, like:
```c
_spawnv(_P_NOWAIT, argv);
```
Probably, we only need to allocate a copy of argv[] (of size argc + 1) and put NULL into the last entry.

---

*[src/test/obj_pool_lock/obj_pool_lock.c, line 48 at r2](https://reviewable.io:443/reviews/pmem/nvml/1477#-KXfDhsGnMbFdE1XbQeX-r2-48:-KXzvOCUza-QMXu1h_Ur:br072uo) ([raw file](https://github.com/pmem/nvml/blob/85b5694cb1427817897da9a33ab7936922b191cd/src/test/obj_pool_lock/obj_pool_lock.c#L48)):*
<details><summary><i>Previously, pbalcer (Piotr Balcer) wrote…</i></summary><blockquote>

This is the third copy of this exact same code. My suggestion would be to move it to unittest.c, and while you are at it, change it so that it automatically retrieves the executable name so that you don't have to do anything else.
</blockquote></details>

:+1: (see the comment above)

---

*[src/test/util_poolset_parse/TEST0, line 45 at r2](https://reviewable.io:443/reviews/pmem/nvml/1477#-KXzvUf4QTvF93agvbUF:-KXzvUf4QTvF93agvbUG:bb7r2f0) ([raw file](https://github.com/pmem/nvml/blob/85b5694cb1427817897da9a33ab7936922b191cd/src/test/util_poolset_parse/TEST0#L45)):*
> ```
> require_test_type medium
> 
> require_build_type debug
> ```

why?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1477)*
<!-- Sent from Reviewable.io -->

wlemkows commented:
Review status: 6 of 11 files reviewed at latest revision, 5 unresolved discussions.

---

*[src/test/blk_pool_lock/blk_pool_lock.c, line 48 at r2](https://reviewable.io:443/reviews/pmem/nvml/1477#-KXzkvuJv4174AbM-sH3:-KYDCmiVRt62Xr4wAOiS:b1x1xrq) ([raw file](https://github.com/pmem/nvml/blob/85b5694cb1427817897da9a33ab7936922b191cd/src/test/blk_pool_lock/blk_pool_lock.c#L48)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary><blockquote>

1) I would just name it "argv".
2) Why not "const"?
3) To be honest, I'd convert into a generic function that would take "int argc, const char *argv[]" and would act as a utility function to spawn a new process with given args (fork + exec equivalent).
</blockquote></details>

I did generic function to create new process using _spawnv. There was needed to add function to append additional parameter, because for Windows we use additional third param in new process. The same function adds NULL at the last elem in spawnv argv.
Reference to other comment, I used _P_WAIT (behavior like on linux). Thanks for it we tests pool open after create by another process.

---

*[src/test/blk_pool_lock/blk_pool_lock.c, line 189 at r2](https://reviewable.io:443/reviews/pmem/nvml/1477#-KXfDhsFyVIvYXnxz-Uv-r2-189:-KYDFcZSWTAgPQWXTqs6:b-xbqpdn) ([raw file](https://github.com/pmem/nvml/blob/85b5694cb1427817897da9a33ab7936922b191cd/src/test/blk_pool_lock/blk_pool_lock.c#L189)):*
<details><summary><i>Previously, pbalcer (Piotr Balcer) wrote…</i></summary><blockquote>

Instead of turning the variable named "path" to an array, add a second argument to those functions.
</blockquote></details>

Because of  I used spawnv to create process I decided to not split this param.

---

*[src/test/util_poolset_parse/TEST0, line 45 at r2](https://reviewable.io:443/reviews/pmem/nvml/1477#-KXzvUf4QTvF93agvbUF:-KYDEK4ruY62BVIvxN2x:b-p5m6ik) ([raw file](https://github.com/pmem/nvml/blob/85b5694cb1427817897da9a33ab7936922b191cd/src/test/util_poolset_parse/TEST0#L45)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary><blockquote>

why?
</blockquote></details>

Because on linux we linked only debug version of libpmemcommon so there is no difference between debug and release version in those tests.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1477)*
<!-- Sent from Reviewable.io -->

pbalcer commented on src/test/unittest/ut.c:
```diff
@@ -593,3 +607,25 @@ ut_checksum(uint8_t *addr, size_t len)
+ * returns increased array and new size by argument
+ */
+char **
+ut_append_to_array(int size, const char *array[],
```
Seems to me you want to grow a buffer and append an element to it.
This is not the way to do it:
1. You might be leaking memory.
2. Without looking at the code, the programmer might think that the array pointer will be reallocated.
3. You can achieve the same thing with a simple realloc(3)
4. No error handling.
5. It's almost never a good idea to allocate memory in one compilation unit, and require the user to free it in another one.

pbalcer commented on src/test/unittest/ut.c:
```diff
@@ -117,6 +117,20 @@ ut_strerror(int errnum, char *buff, size_t bufflen)
+ut_new_process(int argc, const char *argv[])
+{
+	int new_size;
+	char **cmd = ut_append_to_array(argc, argv, NULL, &new_size);
```
Is there a reason you need the last element to be NULL? Please write so in a comment.

pbalcer commented on src/test/obj_pool_lock/obj_pool_lock.c:
```diff
@@ -152,24 +107,31 @@ test_open_in_different_process(const char *path, int sleep)
+test_open_in_different_process(char **argv, int argc, int sleep)
 {
 	PMEMobjpool *pop;
+	char *third_arg = "X";
```
It took me a little while to figure out the meaning and purpose of this 'third argument' variable. Cryptic as hell ;)
If you decide to leave this little trickery here, I suggest adding a comment as to why.

krzycz commented:
Reviewed 5 of 5 files at r3.
Review status: all files reviewed at latest revision, 10 unresolved discussions.

---

*[src/test/blk_pool_lock/blk_pool_lock.c, line 68 at r3](https://reviewable.io:443/reviews/pmem/nvml/1477#-KYJBkMLEcsOWAOsUXCR:-KYJBkMLE9v0v-w3aQK0:bbeftmf) ([raw file](https://github.com/pmem/nvml/blob/151bef9ef57ba49c848d05a6929c82dc26ea31c8/src/test/blk_pool_lock/blk_pool_lock.c#L68)):*
> ```C
> #ifndef _WIN32
> static void
> test_open_in_different_process(char **argv, int argc, int sleep)
> ```

Just a style nit - "argc" usually comes first, then "argv".

---

*[src/test/blk_pool_lock/blk_pool_lock.c, line 128 at r3](https://reviewable.io:443/reviews/pmem/nvml/1477#-KYErH6A52CdmLjg5FGJ:-KYErH6A52CdmLjg5FGK:b2nigcz) ([raw file](https://github.com/pmem/nvml/blob/151bef9ef57ba49c848d05a6929c82dc26ea31c8/src/test/blk_pool_lock/blk_pool_lock.c#L128)):*
> ```C
> 		UT_FATAL("!create");
> 
> 	char **cmd = ut_append_to_array(argc, argv, third_arg, &new_size);
> ```

xxx

---

*[src/test/unittest/unittest.h, line 602 at r3](https://reviewable.io:443/reviews/pmem/nvml/1477#-KYOBh2cDih_nyjkYuq_:-KYOBh2cBvFITZj3JYLw:bj53nea) ([raw file](https://github.com/pmem/nvml/blob/151bef9ef57ba49c848d05a6929c82dc26ea31c8/src/test/unittest/unittest.h#L602)):*
> ```objc
>     ut_pthread_join(__FILE__, __LINE__, __func__, thread, value_ptr)
> 
> /*
> ```

#ifdef _WIN32

---

*[src/test/unittest/ut.c, line 128 at r3](https://reviewable.io:443/reviews/pmem/nvml/1477#-KYD5-9j3XVdhz7Y26TV-r3-128:-KYOAWKm6EEOdIuXMfiY:b-zes8lr) ([raw file](https://github.com/pmem/nvml/blob/151bef9ef57ba49c848d05a6929c82dc26ea31c8/src/test/unittest/ut.c#L128)):*
<details><summary><i>Previously, pbalcer (Piotr Balcer) wrote…</i></summary><blockquote>

Is there a reason you need the last element to be NULL? Please write so in a comment.
</blockquote></details>

I think you should rather call this function once (before a call to _spawn()). Now, ut_append_to_array() is called twice.
Or maybe make it a variadic function:
```c
intptr_t ut_spawn(int argc, const char **argv, int argc2, ...)
{
  char **argv2 = calloc(argc + argc2 + 1, sizeof(char *)); /* 1 for terminating NULL */
  if (argv2 == NULL) {
    ERR(...);
    return -1;
  }
  memcpy(argv2, argv, argc * sizeof(char *));

  va_list ap;
  va_start(ap, argc2);
  for (int i = 0; i < argc2; i++)
    argv2[argc + i] = va_arg(ap, char *);
  va_end(ap);

  intptr_t ret = _spawnv(_P_WAIT, argv2[0], argv2);
  free(argv2);
}
```

Then call it like:
```c
ut_spawn(argc, argv, 1, "x"); 
ut_spawn(argc, argv, 2, "x", "y");
```

You can also iterate thru variadic args to count them first and allocate argv2 and then iterate again to populate argv2;
So, the prototype would be:
```c
intptr_t ut_spawn(int argc, const char **argv, ...);
```

---

*[src/test/util_poolset_parse/TEST0, line 45 at r2](https://reviewable.io:443/reviews/pmem/nvml/1477#-KXzvUf4QTvF93agvbUF:-KYO8zxb3qqMIJu5nl-R:b-l43msf) ([raw file](https://github.com/pmem/nvml/blob/85b5694cb1427817897da9a33ab7936922b191cd/src/test/util_poolset_parse/TEST0#L45)):*
<details><summary><i>Previously, wlemkows (Weronika Lewandowska) wrote…</i></summary><blockquote>

Because on linux we linked only debug version of libpmemcommon so there is no difference between debug and release version in those tests.
</blockquote></details>

Well, there is no technical reason for not using non-debug variant of libpmemcommon in non-debug tests, and we may change it in the future.
Also, libpmemcommon is linked statically, and "require_build_type" applies to other libraries (libpmem, libppmemobj, etc.).
So, I would not change it.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1477)*
<!-- Sent from Reviewable.io -->

wlemkows commented:
Review status: 5 of 10 files reviewed at latest revision, 10 unresolved discussions.

---

*[src/test/blk_pool_lock/blk_pool_lock.c, line 68 at r3](https://reviewable.io:443/reviews/pmem/nvml/1477#-KYJBkMLEcsOWAOsUXCR:-KYTZaUQxfr3pa2TfREk:b-896fix) ([raw file](https://github.com/pmem/nvml/blob/151bef9ef57ba49c848d05a6929c82dc26ea31c8/src/test/blk_pool_lock/blk_pool_lock.c#L68)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary><blockquote>

Just a style nit - "argc" usually comes first, then "argv".
</blockquote></details>

Done.

---

*[src/test/blk_pool_lock/blk_pool_lock.c, line 128 at r3](https://reviewable.io:443/reviews/pmem/nvml/1477#-KYErH6A52CdmLjg5FGJ:-KYXOdZzm_AdnQT5vCC_:b-5zykog) ([raw file](https://github.com/pmem/nvml/blob/151bef9ef57ba49c848d05a6929c82dc26ea31c8/src/test/blk_pool_lock/blk_pool_lock.c#L128)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary><blockquote>

xxx
</blockquote></details>

I removed this function.

---

*[src/test/obj_pool_lock/obj_pool_lock.c, line 117 at r3](https://reviewable.io:443/reviews/pmem/nvml/1477#-KXfDhsGnMbFdE1XbQeX-r3-117:-KYTfbkgL23U9P--cVRQ:b-896fix) ([raw file](https://github.com/pmem/nvml/blob/151bef9ef57ba49c848d05a6929c82dc26ea31c8/src/test/obj_pool_lock/obj_pool_lock.c#L117)):*
<details><summary><i>Previously, pbalcer (Piotr Balcer) wrote…</i></summary><blockquote>

It took me a little while to figure out the meaning and purpose of this 'third argument' variable. Cryptic as hell ;)
If you decide to leave this little trickery here, I suggest adding a comment as to why.
</blockquote></details>

Done.

---

*[src/test/unittest/unittest.h, line 602 at r3](https://reviewable.io:443/reviews/pmem/nvml/1477#-KYOBh2cDih_nyjkYuq_:-KYTirSJnyvC48yMSW02:b-896fix) ([raw file](https://github.com/pmem/nvml/blob/151bef9ef57ba49c848d05a6929c82dc26ea31c8/src/test/unittest/unittest.h#L602)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary><blockquote>

#ifdef _WIN32
</blockquote></details>

Done.

---

*[src/test/unittest/ut.c, line 128 at r3](https://reviewable.io:443/reviews/pmem/nvml/1477#-KYD5-9j3XVdhz7Y26TV-r3-128:-KYYnsVZr8Hj4cCFp0oo:b-896fix) ([raw file](https://github.com/pmem/nvml/blob/151bef9ef57ba49c848d05a6929c82dc26ea31c8/src/test/unittest/ut.c#L128)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary><blockquote>

I think you should rather call this function once (before a call to _spawn()). Now, ut_append_to_array() is called twice.
Or maybe make it a variadic function:
```c
intptr_t ut_spawn(int argc, const char **argv, int argc2, ...)
{
  char **argv2 = calloc(argc + argc2 + 1, sizeof(char *)); /* 1 for terminating NULL */
  if (argv2 == NULL) {
    ERR(...);
    return -1;
  }
  memcpy(argv2, argv, argc * sizeof(char *));

  va_list ap;
  va_start(ap, argc2);
  for (int i = 0; i < argc2; i++)
    argv2[argc + i] = va_arg(ap, char *);
  va_end(ap);

  intptr_t ret = _spawnv(_P_WAIT, argv2[0], argv2);
  free(argv2);
}
```

Then call it like:
```c
ut_spawn(argc, argv, 1, "x"); 
ut_spawn(argc, argv, 2, "x", "y");
```

You can also iterate thru variadic args to count them first and allocate argv2 and then iterate again to populate argv2;
So, the prototype would be:
```c
intptr_t ut_spawn(int argc, const char **argv, ...);
```

</blockquote></details>

Done.

---

*[src/test/unittest/ut.c, line 616 at r3](https://reviewable.io:443/reviews/pmem/nvml/1477#-KYD5-9j3XVdhz7Y26TV-r3-616:-KYXOYeG57UQO4hnPmAL:b-5zykog) ([raw file](https://github.com/pmem/nvml/blob/151bef9ef57ba49c848d05a6929c82dc26ea31c8/src/test/unittest/ut.c#L616)):*
<details><summary><i>Previously, pbalcer (Piotr Balcer) wrote…</i></summary><blockquote>

Seems to me you want to grow a buffer and append an element to it.
This is not the way to do it:
1. You might be leaking memory.
2. Without looking at the code, the programmer might think that the array pointer will be reallocated.
3. You can achieve the same thing with a simple realloc(3)
4. No error handling.
5. It's almost never a good idea to allocate memory in one compilation unit, and require the user to free it in another one.
</blockquote></details>

I removed this function.

---

*[src/test/util_poolset_parse/TEST0, line 45 at r2](https://reviewable.io:443/reviews/pmem/nvml/1477#-KXzvUf4QTvF93agvbUF:-KYU4YoOklxEHMt1CM-n:b-896fix) ([raw file](https://github.com/pmem/nvml/blob/85b5694cb1427817897da9a33ab7936922b191cd/src/test/util_poolset_parse/TEST0#L45)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary><blockquote>

Well, there is no technical reason for not using non-debug variant of libpmemcommon in non-debug tests, and we may change it in the future.
Also, libpmemcommon is linked statically, and "require_build_type" applies to other libraries (libpmem, libppmemobj, etc.).
So, I would not change it.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1477)*
<!-- Sent from Reviewable.io -->

plebioda commented:
Reviewed 5 of 9 files at r1, 6 of 6 files at r4.
Review status: all files reviewed at latest revision, 15 unresolved discussions.

---

*[src/test/blk_pool_lock/blk_pool_lock.c, line 75 at r4](https://reviewable.io:443/reviews/pmem/nvml/1477#-KZ36BAoXNPzSOojIgGu:-KZ36BAp4BTZ9-egB5D6:b-qmkwf) ([raw file](https://github.com/pmem/nvml/blob/0d1e44907bff1371499ffcb1c330f3e84682a953/src/test/blk_pool_lock/blk_pool_lock.c#L75)):*
> ```C
> 	if (pid < 0)
> 		UT_FATAL("fork failed");
> ```

char *path = argv[1]; ?

It is better if the arguments would change somehow in the future.

---

*[src/test/blk_pool_lock/blk_pool_lock.c, line 119 at r4](https://reviewable.io:443/reviews/pmem/nvml/1477#-KZ36N3c0uHh8RwbiD2G:-KZ36N3c0uHh8RwbiD2H:bef9okt) ([raw file](https://github.com/pmem/nvml/blob/0d1e44907bff1371499ffcb1c330f3e84682a953/src/test/blk_pool_lock/blk_pool_lock.c#L119)):*
> ```C
> 	if (sleep > 0)
> 		return;
> ```

char *path = argv[1];

---

*[src/test/log_pool_lock/log_pool_lock.c, line 80 at r4](https://reviewable.io:443/reviews/pmem/nvml/1477#-KZ36h4onqjruSj6wBo3:-KZ36h4onqjruSj6wBo4:b3kql) ([raw file](https://github.com/pmem/nvml/blob/0d1e44907bff1371499ffcb1c330f3e84682a953/src/test/log_pool_lock/log_pool_lock.c#L80)):*
> ```C
> 		if (sleep)
> 			usleep(sleep);
> 		while (access(argv[1], R_OK))
> ```

"

---

*[src/test/log_pool_lock/log_pool_lock.c, line 120 at r4](https://reviewable.io:443/reviews/pmem/nvml/1477#-KZ36jnHHArSb_fmc2Ji:-KZ36jnHHArSb_fmc2Jj:b3kql) ([raw file](https://github.com/pmem/nvml/blob/0d1e44907bff1371499ffcb1c330f3e84682a953/src/test/log_pool_lock/log_pool_lock.c#L120)):*
> ```C
> 
> 	/* before starting the 2nd process, create a pool */
> 	log = pmemlog_create(argv[1], PMEMLOG_MIN_POOL, S_IWUSR | S_IRUSR);
> ```

"

---

*[src/test/obj_pool_lock/obj_pool_lock.c, line 81 at r4](https://reviewable.io:443/reviews/pmem/nvml/1477#-KZ36rty99eenVJwK7ye:-KZ36rty99eenVJwK7yf:b3kql) ([raw file](https://github.com/pmem/nvml/blob/0d1e44907bff1371499ffcb1c330f3e84682a953/src/test/obj_pool_lock/obj_pool_lock.c#L81)):*
> ```C
> 		if (sleep)
> 			usleep(sleep);
> 		while (access(argv[1], R_OK))
> ```

"

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1477)*
<!-- Sent from Reviewable.io -->

wlemkows commented:
Review status: 6 of 10 files reviewed at latest revision, 12 unresolved discussions.

---

*[src/test/blk_pool_lock/blk_pool_lock.c, line 75 at r4](https://reviewable.io:443/reviews/pmem/nvml/1477#-KZ36BAoXNPzSOojIgGu:-KZ5TnbQdF4r3zjP2RX9:b-896fix) ([raw file](https://github.com/pmem/nvml/blob/0d1e44907bff1371499ffcb1c330f3e84682a953/src/test/blk_pool_lock/blk_pool_lock.c#L75)):*
<details><summary><i>Previously, plebioda wrote…</i></summary><blockquote>

char *path = argv[1]; ?

It is better if the arguments would change somehow in the future.
</blockquote></details>

Done.

---

*[src/test/blk_pool_lock/blk_pool_lock.c, line 119 at r4](https://reviewable.io:443/reviews/pmem/nvml/1477#-KZ36N3c0uHh8RwbiD2G:-KZ5Tmg4fLQHkMJ4UkF_:b-896fix) ([raw file](https://github.com/pmem/nvml/blob/0d1e44907bff1371499ffcb1c330f3e84682a953/src/test/blk_pool_lock/blk_pool_lock.c#L119)):*
<details><summary><i>Previously, plebioda wrote…</i></summary><blockquote>

char *path = argv[1];
</blockquote></details>

Done.

---

*[src/test/log_pool_lock/log_pool_lock.c, line 80 at r4](https://reviewable.io:443/reviews/pmem/nvml/1477#-KZ36h4onqjruSj6wBo3:-KZ5URZuk02gkkmU7M7K:b-896fix) ([raw file](https://github.com/pmem/nvml/blob/0d1e44907bff1371499ffcb1c330f3e84682a953/src/test/log_pool_lock/log_pool_lock.c#L80)):*
<details><summary><i>Previously, plebioda wrote…</i></summary><blockquote>

"
</blockquote></details>

Done.

---

*[src/test/log_pool_lock/log_pool_lock.c, line 120 at r4](https://reviewable.io:443/reviews/pmem/nvml/1477#-KZ36jnHHArSb_fmc2Ji:-KZ5UVIwbzc_8AXUrgWM:b-896fix) ([raw file](https://github.com/pmem/nvml/blob/0d1e44907bff1371499ffcb1c330f3e84682a953/src/test/log_pool_lock/log_pool_lock.c#L120)):*
<details><summary><i>Previously, plebioda wrote…</i></summary><blockquote>

"
</blockquote></details>

Done.

---

*[src/test/obj_pool_lock/obj_pool_lock.c, line 81 at r4](https://reviewable.io:443/reviews/pmem/nvml/1477#-KZ36rty99eenVJwK7ye:-KZ5XN1GpiYEAEfAN1j2:b-896fix) ([raw file](https://github.com/pmem/nvml/blob/0d1e44907bff1371499ffcb1c330f3e84682a953/src/test/obj_pool_lock/obj_pool_lock.c#L81)):*
<details><summary><i>Previously, plebioda wrote…</i></summary><blockquote>

"
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1477)*
<!-- Sent from Reviewable.io -->

plebioda commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 4 of 4 files at r5.
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1477#-:-KZ5zzOJuMWWR5eIQc0q:bnfp4nl)*
<!-- Sent from Reviewable.io -->

krzycz commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 2 of 6 files at r4, 4 of 4 files at r5.
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1477#-:-KZ6qBgyz1qy760N_NCl:bnfp4nl)*
<!-- Sent from Reviewable.io -->
